src/Factory/Tags.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Factory;
+
+use App\Entity\Tag;
+use App\Repository\TagRepository;
+
+final class Tags
+{
+    public function __construct(
+        private readonly TagRepository $tagRepository
+    ) {
+    }
+
+    /**
+     * Create tag or find matching in the database.
+     *
+     * @param array<string> $tagsName
+     *   The tag names to create/lockup in the database as strings
+     *
+     * @return iterable<Tag>
+     *   Yield tag entities from the database
+     */
+    public function createOrLookup(array $tagsName): iterable
+    {
+        foreach ($tagsName as $tagName) {
+            $tag = $this->tagRepository->findOneBy(['name' => $tagName]);
+            if (is_null($tag)) {
+                $tag = new Tag();
+                $tag->setName($tagName);
+                $this->tagRepository->save($tag);
```

Comment:
Have the tag names been normalized to be unique? Otherwise you should flush here to make sure that you don't create duplicate tags if `['hest', 'hest']`, say, is passed to this function.

Reply:
They have been normalized at this point and should not contain repetitions.